src/main/webapp/components/workspace/hooks/use-save-workspace.tsx

```diff
@@ -17,11 +17,15 @@ export default () => {
     }
   `
   const [save] = useMutation(mutation)
-  return (workspace: Workspace) => {
+  return async (
```

Comment:
While I agree this hook should be made async, so that you can manage state of requests, I don't think that we should be passing in a function to update the loading state.  I think the workspace should be handling the state on it's own.

Reply:
I was considering both ways. I decided to do it this way so the saving icon is always shown whenever the hook is called. That way devs in the future don't have to worry about setting the saving state.

Reply:
I think what @willwill96 suggested is the better option. You don't need to worry about setting the saving state if you're just passing it back through the hook. You can return [saveWorkspace, saving] from the hook, and the component that needs the saving state could render based on the saving variable.